tools/pythonpkg/tests/fast/test_filesystem.py

```diff
@@ -202,6 +202,6 @@ def test_read_hive_partition(self, duckdb_cursor: DuckDBPyConnection, memory: Ab
         with memory.open('/root/a=1/data_0.csv', 'wb') as fh:
             fh.write(b'1\n')
 
-        duckdb_cursor.execute('''SELECT * FROM read_csv_auto('memory://root/*/*.csv', HIVE_PARTITIONING = 1);''')
```

Comment:
The [error](https://github.com/lverdoes/duckdb/actions/runs/5056496165/jobs/9074036883#step:7:769) 
Because of the autocasting of partition types: the `'1'` (VARCHAR) is converted to a `1` (INTEGER), but the assertion is still expecting read a '1'.
Disabling autocast makes sure the '1' stays a VARCHAR.

Reply:
Ok so that is expected and correct behaviour then. Maybe instead update the assertion, or test both?